Scenarios/CGCT/CGCT/DemoScenarioImages.m

```diff
@@ -0,0 +1,94 @@
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#import "DemoScenarioImages.h"
+#import <CoreGraphics/CoreGraphics.h>
+#import <CoreText/CoreText.h>
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+
+@implementation DemoScenarioImages {
+    UIImage* patternImage;
+};
+
+static NSString* patternName;
+static CGBlendMode blendModes[] = { kCGBlendModeNormal,    kCGBlendModeScreen,     kCGBlendModeDarken,    kCGBlendModeHue,
+                                    kCGBlendModeXOR,       kCGBlendModeSaturation, kCGBlendModeOverlay,   kCGBlendModeSoftLight,
+                                    kCGBlendModeHardLight, kCGBlendModeLuminosity, kCGBlendModeDifference };
+
+- (NSString*)name {
+    return @"Images";
+}
+
+- (UIColor*)backgroundColor {
+    return [UIColor colorWithRed:.72 green:.6 blue:.4 alpha:1];
+}
+
+- (instancetype)init {
+    if (self = [super init]) {
+        patternImage = [UIImage imageWithContentsOfFile:[[NSBundle mainBundle] pathForResource:@"sunset" ofType:@"png"]];
+    }
+    return self;
+}
+
+- (void)drawDemoIntoContext:(CGContextRef)context withFrame:(CGRect)bounds {
+    CGContextScaleCTM(context, 1.0f, -1.0f);
+    CGContextTranslateCTM(context, bounds.size.width * .03, -bounds.size.height * .98);
+    CGContextSaveGState(context);
+
+    CGFloat aspectRatio = patternImage.size.height / patternImage.size.width;
+    CGFloat catSize = bounds.size.width / 3.75;
+    CGContextSetRGBFillColor(context, 1, 0, 0, .3);
+    int colorIndex = 0;
+
+    CGFloat fillColors[48];
+    for (CGFloat r = .4; r < 1; r += .4) {
+        for (CGFloat g = .4; g < 1; g += .4) {
+            for (CGFloat b = .4; b < 1; b += .4) {
+                for (CGFloat a = .4; a < 1; a += .4) {
+                    fillColors[colorIndex] = r;
+                    fillColors[colorIndex + 1] = g;
+                    fillColors[colorIndex + 2] = b;
+                    fillColors[colorIndex + 3] = a;
+                    colorIndex += 4;
+                }
+            }
+        }
+    }
+
+    int blendModeIndex = 0;
+    for (CGFloat x = 0; x < 1; x += .34) {
+        for (CGFloat y = 0; y < 1; y += .25) {
+            CGContextSetRGBFillColor(context,
+                                     fillColors[blendModeIndex * 4],
+                                     fillColors[blendModeIndex * 4 + 1],
+                                     fillColors[blendModeIndex * 4 + 2],
+                                     fillColors[blendModeIndex * 4 + 3]);
+            CGRect pane = CGRectMake(x * bounds.size.width, y * bounds.size.height, catSize, catSize * aspectRatio);
+            CGContextDrawImage(context, pane, patternImage.CGImage);
+
+            CGContextSetBlendMode(context, blendModes[blendModeIndex]);
```

Comment:
also, the GState is never being saved, Each successive image is being drawn in the previous color plate's blend mode ...